### ISSUE TITLE
Add tests for `data` set to Buffer and ArrayBuffer

### ISSUE THREAD

**Issue by gr2m** [octokit/request.js#110]
compare test from `@octokit/rest` v16

```js
const stringToArrayBuffer = require("string-to-arraybuffer");
const { getInstance } = require("../util");

require("../mocha-node-setup");

describe("api.github.com", () => {
  let octokit;

  beforeEach(() => {
    return getInstance("release-assets", {
      auth: "token 0000000000000000000000000000000000000001"
    }).then(instance => {
      octokit = instance;
    });
  });

  it("octokit.request with Buffer data", () => {
    return octokit.repos
      .getReleaseByTag({
        owner: "octokit-fixture-org",
        repo: "release-assets",
        tag: "v1.0.0"
      })

      .then(result => {
        return octokit.request({
          method: "POST",
          url: result.data.upload_url,
          headers: {
            "content-type": "text/plain"
          },
          data: Buffer.from("Hello, world!\n"),
          name: "test-upload.txt",
          label: "test"
        });
      });
  });

  it("octokit.repos.uploadReleaseAsset as ArrayBuffer", () => {
    return octokit.repos
      .getReleaseByTag({
        owner: "octokit-fixture-org",
        repo: "release-assets",
        tag: "v1.0.0"
      })

      .then(result => {
        return octokit.request({
          method: "POST",
          url: result.data.upload_url,
          headers: {
            "content-type": "text/plain"
          },
          data: stringToArrayBuffer("Hello, world!\n"),
          name: "test-upload.txt",
          label: "test"
        });
      });
  });
});
```

This is part of https://github.com/octokit/rest.js/issues/1546

**Comment by romanbalayan** [comment]
Hi @gr2m,

Also took a stab at this! Also quite unsure as well if I captured the essence here. Also had to add a new dev dependency - "string-to-arraybuffer"

Will be glad to hear your feedback! Thanks!

**Comment by wolfy1339** [comment]
Fixed by #438
